core/src/main/java/org/apache/accumulo/core/iterators/IteratorUtil.java

```diff
@@ -449,8 +449,9 @@ public static IteratorConfig toIteratorConfig(List<IteratorSetting> iterators) {
 
   public static SortedKeyValueIterator<Key,Value> setupSystemScanIterators(
       SortedKeyValueIterator<Key,Value> source, Set<Column> cols, Authorizations auths,
-      byte[] defaultVisibility) throws IOException {
-    DeletingIterator delIter = new DeletingIterator(source, false);
+      byte[] defaultVisibility, AccumuloConfiguration conf) throws IOException {
+    SortedKeyValueIterator<Key,Value> delIter = DeletingIterator.wrap(source, false,
+        DeletingIterator.getBehavior(conf));
```

Comment:
Just a sanity check: is "conf" coming from the table configuration everywhere `setupSystemScanIterators` is called?

Reply:
I think so.